browser-test/src/end_to_end_enumerators.test.ts

```diff
@@ -0,0 +1,135 @@
+    await page.click('#add-block-button');
+    await page.click('button:text("enumerator-ete-enumerator")');
+    await page.click('#create-repeated-block-button');
+    expect(await page.innerText('id=question-bank-questions')).toContain('enumerator-ete-enumerator-name');
+    expect(await page.innerText('id=question-bank-questions')).toContain('enumerator-ete-enumerator-enumerator');
+
+    // Create the rest of the program.
+    // Add repeated name question
+    await page.click('button:text("enumerator-ete-enumerator-name")');
+
+    // Create another repeated block and add the nested enumerator question
+    await page.click('p:text("Block 2")');
+    await page.click('#create-repeated-block-button');
+    await page.click('button:text("enumerator-ete-enumerator-enumerator")');
+
+    // Create a nested repeated block and add the nested text question
+    await page.click('#create-repeated-block-button');
+    await page.click('button:text("enumerator-ete-enumerator-enumerator-text")');
+
+    // Publish!
+    await adminPrograms.publishProgram(programName);
+
+    await endSession(browser);
+  });
+
+
+  it('Applicant can fill in lots of blocks, and then go back and delete some repeated entities', async () => {
+    const { browser, page } = await startSession();
+    await loginAsGuest(page);
+    await selectApplicantLanguage(page, 'English');
+    const applicantQuestions = new ApplicantQuestions(page);
+    await applicantQuestions.applyProgram(programName);
+
+    // Fill in name question
+    await applicantQuestions.answerNameQuestion("first name", "last name");
+    await applicantQuestions.saveAndContinue();
+
+    // Put in two things in the enumerator question
+    await applicantQuestions.addEnumeratorAnswer("enum one");
+    await applicantQuestions.addEnumeratorAnswer("enum two");
+    await applicantQuestions.saveAndContinue();
+
+    // FIRST REPEATED ENTITY
+    // Answer name
+    await applicantQuestions.answerNameQuestion("enum one first", "enum one last");
+    await applicantQuestions.saveAndContinue();
+
+    // Put nothing in the first nested enumerator
+    await applicantQuestions.saveAndContinue();
+
+    // SECOND REPEATED ENTITY
+    // Answer name
+    await applicantQuestions.answerNameQuestion("enum two first", "enum two last");
+    await applicantQuestions.saveAndContinue();
+
+    // Put two things in the second nested enumerator
+    await applicantQuestions.addEnumeratorAnswer("thing one");
+    await applicantQuestions.addEnumeratorAnswer("thing two");
+    await applicantQuestions.saveAndContinue();
+
+    // Answer two nested repeated text questions
+    await applicantQuestions.answerTextQuestion("hello");
+    await applicantQuestions.saveAndContinue();
+    await applicantQuestions.answerTextQuestion("world");
+    await applicantQuestions.saveAndContinue();
+
+    // Make sure the enumerator answers are in the review page
+    expect(await page.innerText("body")).toContain("first name");

```

Comment:
Nit. I feel like we should have a helper function for this. (I'm surprised jest doesn't already.)

`expectContains(body, ["first name", "last name"...]);`

Reply:
There's https://github.com/playwright-community/expect-playwright#why-do-i-need-it but I don't want to add dependencies...